src/bp/core/repositories/events.ts

```diff
@@ -0,0 +1,64 @@
+import * as sdk from 'botpress/sdk'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+
+import Database from '../database'
+import { TYPES } from '../types'
+
+export interface EventRepository {
+  findEvents(fields: Partial<sdk.IO.StoredEvent>, searchParams: sdk.EventSearchParams)
+  deleteBeforeDate(date: Date): Promise<void>
+}
+
+export const DefaultSearchParams: sdk.EventSearchParams = {
```

Comment:
We need a fromDate & toDate filter as well

Reply:
I've added the base elements, we can extend it for different requirements easily by adding props in the sdk and handling them in the repository